Backport "Added missing type names to uri parsing regex for memory provider" ltr

domi4484 opened the pull request:
Backport of #44099 and #43916

sebastic committed:
Add support for PROJ 8.0.0 in FindProj.cmake.

proj_api.h is no longer available in PROJ 8.0.0, use proj.h instead.

Fixes: #41735

domi4484 committed:
MemoryProvider native type is boolean and not bool

domi4484 committed:
Fix test

domi4484 committed:
Added missing type names to uri parsing regex for memory provider

- This was causing strange field names when cloning layers. fixes #44087
- Some code cleanup + QRegexp replaced with QRegularExpression

domi4484 committed:
Update src/core/providers/memory/qgsmemoryprovider.cpp

Co-authored-by: Andrea Giudiceandrea <andreaerdna@libero.it>

domi4484 committed:
MemoryProvider respect field name when parsing a URI with unknown typeName

domi4484 committed:
Default length/precision for double/real fields

domi4484 committed:
Regex as cons thread_local

nyalldawson committed:
Rework API and improve memory handling of graph objects

Instead of QgsGraphBuilder::graph() taking the ownership of the graph
and leaving the QgsGraphBuilder in an unpredictable state, add an
explicit "takeGraph" method which takes the existing graph and
make the existing "graph" method just return a copy of the graph.

Also fix corresponding memory leaks in network analysis processing
algorithms.

Fixes #44079

(cherry picked from commit 404094776cf09ef9a5f7ad1800390b721c972f80)

nyalldawson committed:
Fix crash in vector layer iteration on Qt6 builds

Decrementing an iterator which is already at the beginning results
in a crash

(cherry picked from commit 6776b1b0d79450ed468679059fa507218c5e9a14)

rouault committed:
QgsMessageLogViewer: limit number of messages to avoid blowing RAM (fixes #44202)

rouault committed:
QgsProcessingAlgorithmDialogBase: limit number of info messages to avoid blowing RAM (fixes #44202)

rouault committed:
QgsProcessingFeedback: limit number of messages to avoid blowing RAM (fixes #44202)

nyalldawson committed:
[ogr] Report ESRI multipatch GDB layers as multipolygonz layers

Apply the same workaround we use for multipatch shapefiles to
GDB sources

(cherry picked from commit 7b283b33aec629a479f62f45cf4b3f7f93f534c1)

nyalldawson committed:
Cleaner loop

(cherry picked from commit 6527d59118fff87b4fafa8e7909f22de48f3bf76)

domi4484 committed:
readd lines lost during merge

domi4484 committed:
Fix tests

nyalldawson committed:
Fix incomplete message bar messages when exporting items from model
designer

(cherry picked from commit 1b9295195796a4a2d8319aad3e50f906abcf9b38)

nyalldawson committed:
Fix leaked GDAL dataset handle when deleting layers

Prevents the deletion from actually occurring until QGIS is closed

(cherry picked from commit 842485f7514f7f334a1cd8a43609c032cd98c3a5)

domi4484 committed:
Fix list types

agiudiceandrea commented on src/core/providers/memory/qgsmemoryprovider.cpp:
```diff
@@ -103,104 +103,117 @@ QgsMemoryProvider::QgsMemoryProvider( const QString &uri, const ProviderOptions
                   // blob
                   << QgsVectorDataProvider::NativeType( tr( "Binary object (BLOB)" ), QStringLiteral( "binary" ), QVariant::ByteArray )
 
+                  // list types
+                  << QgsVectorDataProvider::NativeType( tr( "String list" ), QStringLiteral( "stringlist" ), QVariant::StringList, 0, 0, 0, 0, QVariant::String )
+                  << QgsVectorDataProvider::NativeType( tr( "Integer list" ), QStringLiteral( "integerlist" ), QVariant::List, 0, 0, 0, 0, QVariant::Int )
+                  << QgsVectorDataProvider::NativeType( tr( "Decimal (real) list" ), QStringLiteral( "doublelist" ), QVariant::List, 0, 0, 0, 0, QVariant::Double )
+                  << QgsVectorDataProvider::NativeType( tr( "Integer (64bit) list" ), QStringLiteral( "doublelist" ), QVariant::List, 0, 0, 0, 0, QVariant::Double )
+
```
See also https://github.com/qgis/QGIS/pull/44446

domi4484 replied:
@agiudiceandrea a more general "list" type was already supported although not an "official" native type already before #42421. Because I have quite changed the implementation and relying on the native types to decide if a field type is supported or not this "old" list types does not work anymore.
Maybe at this point I should revert all of this and just add the missing types in the regex (for ltr)

3nids committed:
fix building of process without bindings

rouault committed:
Merge pull request #44462 from qgis/backport-44449-to-queued_ltr_backports

[Backport queued_ltr_backports] Fix leaked GDAL dataset handle when deleting layers (3.20)

nyalldawson committed:
Initialize return buffers in GetRawChar8 et al. (Port DWG read fix
from https://github.com/LibreCAD/libdxfrw)

From the upstream message:
GetRawChar8() would return spurious memory contents when the read
failed,leading to weird crashes later in the code.
Seen with older non-Autocad DWG files (that Autodesk DWG Trueview
however accepts with just a warning about their foreign origin)
obtained from RC model plane enthusiasts' free plans archive
aerofred.com, e.g. their "Canadair p5.dwg"

https://github.com/LibreCAD/libdxfrw/commit/082409c03dfb90ac02c8a9354189206f2a771951

nyalldawson committed:
Fix indentation issue

troopa81 committed:
Fix handleSSlError ignore saved errors

rduivenvoorde committed:
Textual fix

rduivenvoorde committed:
Fix for not resetting numberOf(left)Segments when switching scalebar unit

uclaros committed:
Fix sticky buttons in expression builder

Samweli committed:
fix singleFile property issue

Samweli committed:
set singleFile property inside the QgsLayoutAtlas constructor

domi4484 committed:
Removed list types as not intended for backport

nyalldawson committed:
[expressions] Don't cache aggregate results if there's an evaluation error.

Otherwise subsequent evaluations of the expression will happily
grab the cached null variant, unaware that it's actually a
result of a broken expression and not a valid null aggregate
calculation.

(cherry picked from commit 97f024206bc1c725e979ac14eb8b6e539c5671e0)

nyalldawson committed:
Raise IndexErrors instead of crashing when an invalid index is
passed to QgsAttributeTableConfig methods from Python

nyalldawson committed:
Sipify

nirvn committed:
[aggregate calculator] Improve result type detection to fix bogus typeless scenarios

nyalldawson committed:
sipify

nyalldawson commented:
@domi4484 can you rebase and force push this branch? The CI tests aren't being triggered for some reason here.

domi4484 commented:
@nyalldawson thank you, I have pushed and rebased and the tests are been executed, but are still failing by TestQgs3DRendering...

agiudiceandrea committed:
[layout] Fix max grid interval value

Increase the maximum value for mIntervalXSpinBox, mIntervalYSpinBox, mOffsetXSpinBox and mOffsetYSpinBox

m-kuhn commented:
It looks like this test shouldn't be run

https://github.com/qgis/QGIS/blob/master/tests/src/3d/testqgs3drendering.cpp#L343-L344

agiudiceandrea committed:
[layout] Allow negative values for grid offset

Add the "minimum" property to mOffsetXSpinBox and mOffsetYSpinBox in order to allow the use of negative values for the map grid offset.

domi4484 commented:
@m-kuhn @3nids these test is not disabled in the queued backports branch. Should I disable it? But are other backports not facing the same problem?

m-kuhn commented:
Flaky tests shouldn't be running in this branch.
You've got a point that it's interesting that it continues to fail in this specific PR I cannot offer an explanation.

domi4484 commented:
@m-kuhn what i meant with "The test is not disabled" is that in the queued ltr branch these two lines are missing:
```cpp
  if ( !QgsTest::runFlakyTests() )
    QSKIP( "This test is flaky and disabled on CI" );
```
Here: https://github.com/qgis/QGIS/blob/84f5d18350cf66acada5e35ee4cdeea192544e25/tests/src/3d/testqgs3drendering.cpp#L328

m-kuhn commented:
@domi4484 can you cherry-pick ce812634ee67b1ba51fe4bce635ee79e953f7ded ?
